libs/execution_context/ebpf_maps.c

```diff
@@ -2922,10 +2922,12 @@ ebpf_map_get_next_key_and_value_batch(
 
         memcpy(key_and_value + output_length + key_size, next_value, value_size);
 
-        if (flags & EBPF_MAP_FIND_FLAG_DELETE) {
-            // If the caller requested deletion, delete the entry.
+        if ((flags & EBPF_MAP_FIND_FLAG_DELETE) && (previous_key != NULL)) {
+            // If the caller requested deletion, delete the previous entry.
             result = ebpf_map_metadata_tables[map->ebpf_map_definition.type].delete_entry(map, previous_key);
             if (result != EBPF_SUCCESS) {
+                EBPF_LOG_MESSAGE_UINT64(
```

Comment:
I would use `EBPF_LOG_FUNCTION_ERROR` instead; or change the return statement of the function to use `EBPF_RETURN_RESULT` macro. The latter will have some perf implications though on the success path.

Reply:
> EBPF_LOG_FUNCTION_ERROR

This macro just returns generic error, rather than "Failed to delete entry" message. It is better to have granular error message.

```
TraceLoggingString(__FUNCTION__ " returned error", "ErrorMessage"),                                       \
TraceLoggingLong(result, "Error"));  
```